iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -17,32 +17,41 @@
 
 #include "iceoryx_hoofs/cxx/helplets.hpp"
 #include "iceoryx_hoofs/testing/timing_test.hpp"
+#include "iceoryx_hoofs/testing/watch_dog.hpp"
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
 #include "iceoryx_posh/popo/untyped_publisher.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
 #include "iceoryx_posh/runtime/posh_runtime.hpp"
 #include "iceoryx_posh/runtime/service_discovery.hpp"
 #include "iceoryx_posh/testing/roudi_gtest.hpp"
 #include "mocks/posh_runtime_mock.hpp"
 #include "test.hpp"
 
-
 #include <type_traits>
 
 namespace
 {
 using namespace ::testing;
 using namespace iox::runtime;
 using namespace iox::cxx;
+using namespace iox::popo;
+using namespace iox::capro;
 using iox::capro::IdString_t;
 using iox::capro::ServiceDescription;
 using iox::roudi::RouDiEnvironment;
 using iox::runtime::ServiceContainer;
 
+
 class ServiceDiscovery_test : public RouDi_GTest
 {
   public:
     void SetUp() override
     {
         searchResultOfFindServiceWithFindHandler.clear();
+        callbackWasCalled = false;
+        serviceContainer.clear();
+        m_watchdog.watchAndActOnFailure([] { GTEST_FAIL(); });
```

Comment:
This is wrong. The watchdog should terminate the process when it runs into a deadlock to avoid blocking the CI for 6 hours.
What you are doing here is just fail the test and block the CI for 6 hours.

Please add a `std::terminate`.